### PR TITLE
Last functions

### PR DESCRIPTION
**Pull request by dariuskrtn**

### CODE REVIEW

**Comment by sauliusgrigaitis** [comment]
A lot of helper functions operate on BeaconState. What do you think about the idea to turn helper functions into methods?

File: helper_functions/helper_functions_interface/src/crypto.rs [hunk 1]
```diff
@@ -1,3 +1,43 @@
+use bls::{AggregatePublicKey, PublicKey, PublicKeyBytes, SignatureBytes};
+use ssz::DecodeError;
+use types::primitives::Domain;
+
+// ok
 pub fn hash(_input: &[u8]) -> Vec<u8> {
```
Comment (sauliusgrigaitis) [inline helper_functions/helper_functions_interface/src/crypto.rs:6]: Shouldn't it return `H256`

File: helper_functions/helper_functions_interface/src/math.rs [hunk 2]
```diff
@@ -5,7 +5,7 @@ pub fn integer_squareroot(_n: u64) -> u64 {
     1
 }
 
-// there's a comment in PR about creating some struct for this
+// ok
 pub fn xor(_bytes_1: &[u8], _bytes_2: &[u8]) -> Result<Vec<u8>, Error> {
```
Comment (sauliusgrigaitis) [inline helper_functions/helper_functions_interface/src/math.rs:9]: Shouldn't it accept `H256` and return `H256`

**Comment by sauliusgrigaitis** [comment]
[is_valid_indexed_attestation](https://github.com/dariuskrtn/framework/blob/40f6387ac753fb102dfa650f0a92b128347edf84/helper_functions/helper_functions_interface/src/predicates.rs#L28) should return boolean?

**Comment by sim31** [comment]
> A lot of helper functions operate on BeaconState. What do you think about the idea to turn helper functions into methods?

Would make sense. But then I think we wouldn't be able to have helper functions in a separate crate.

> [is_valid_indexed_attestation](https://github.com/dariuskrtn/framework/blob/40f6387ac753fb102dfa650f0a92b128347edf84/helper_functions/helper_functions_interface/src/predicates.rs#L28) should return boolean?

The idea is to return an Error describing why the attestation is invalid instead of just boolean false. An empty tuple is returned instead of true. That's how lighthouse does it as well.

**Review by sauliusgrigaitis — COMMENTED** [review]
> The idea is to return an Error describing why the attestation is invalid instead of just boolean false. An empty tuple is returned instead of true. That's how lighthouse does it as well.

The question is more about the fact that the function name starts with `is_` so it should return `boolean`. If it doesn't return `boolean` then probably the name should not start with `is_`

**Pull request merged**